genienlp/data_utils/bootleg.py

```diff
@@ -198,35 +196,35 @@ def add_type_tokens(self, sentence, features):
         assert len(sentence_tokens) == len(features)
         sentence_plus_types_tokens = []
         i = 0
-        if self.args.add_types_to_text == 'insert' or self.args.add_qids_to_text == 'insert':
+        if self.args.add_entities_to_text == 'insert':
             while i < len(sentence_tokens):
                 token = sentence_tokens[i]
                 feat = features[i]
                 # token is an entity
-                if any([val != self.args.ned_features_default_val[0] for val in feat.type_id]):
+                if any([val != 0 for val in feat.type_id]):
                     final_token = '<e> '
                     final_types = ''
-                    if self.args.add_types_to_text != 'no':
+                    if 'type_id' in self.args.entity_attributes:
                         if self.bootleg_post_process_types:
                             all_types = ' | '.join(
-                                set(
-                                    self.qid2almond_type[self.id2typeqid[id]]
+                                sorted(
```

Comment:
Do you need both "set" and "sorted" to remove duplicates?

Reply:
`set()` is not useful here since the additional types are already omitted at this point. The set behavior is here: https://github.com/stanford-oval/genienlp/blob/49ab643afd608a60de22b27e804d3f9d215f91f6/genienlp/data_utils/bootleg.py#L345
to avoid appending similar types before trimming the extra ones.